### PR TITLE
do_after progress bars

### PR DESCRIPTION
**Pull request by Lohikar**
Adds the do_after progress bars that were originally going to be added with lighting, but got scrapped due to PR-bloat.

changes:
- `do_after` actions now have a progress bar.
- Added a changelog for directional lighting as I apparently forgot it.

The `do_after` proc's required arguments are unchanged, however it now accepts the argument `act_target`, which allows the caller to specify what object the progress bar should appear over. Defaults to user.

### CODE REVIEW

File: code/_helpers/unsorted.dm [hunk 1]
```diff
@@ -646,45 +646,109 @@ proc/GaussRandRound(var/sigma,var/roundto)
...
+	if(user && user.client)
+		if(!progbar)
+			progbar = image(icon = 'icons/effects/doafter_icon.dmi', loc = target, icon_state = "prog_bar_0")
+			progbar.layer = 21
+			progbar.pixel_z = WORLD_ICON_SIZE
+			progbar.appearance_flags = RESET_TRANSFORM
+	for (var/i = 1 to numticks)
+		if(user && user.client && progbar)
+			progbar.icon_state = "prog_bar_[round(((i / numticks) * 100), 10)]"
+			user.client.images |= progbar
```
Comment (NanakoAC) [inline code/_helpers/unsorted.dm]: |= is kind of expensive, is it necessary?

File: html/changelogs/lohikar-lighting.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 author: Lohikar
 delete-after: True
 changes:
```
Comment (NanakoAC) [inline html/changelogs/lohikar-lighting.yml]: this stuff doesnt seem to have anything to do with doafters. I think you got some extraneous changelogs

**Comment by NanakoAC** [comment]
The doafter DMI file only seems to have ten iconstates. Could you perhaps add a few more inbetween stages for extra granularity?

File: html/changelogs/lohikar-lighting.yml (hunk 2 above)
Comment (Lohikar) [inline html/changelogs/lohikar-lighting.yml]: If you read the PR summary, you'll see that I forgot the directional lighting changelog in the first place and was re-adding it.
The other one was just a tweak to the existing lighting changelog.

100% intentional.

**Comment by Lohikar** [comment]
I do not particularly feel like trying my hand at spriting to add more icon states to make it smoother, so it'll stay at its current level of granularity.
I don't feel that the bar should be hidden for brief actions, and no it is currently not possible to have it not show.

**Pull request merged**